File: ios/vendored/sdk50/@stripe/stripe-react-native/ios/CardFormManager.m
#import <Foundation/Foundation.h>
#import <ABI50_0_0React/ABI50_0_0RCTBridgeModule.h>
#import <ABI50_0_0React/ABI50_0_0RCTViewManager.h>

@interface ABI50_0_0RCT_EXTERN_REMAP_MODULE(CardFormManager, ABI50_0_0CardFormManager, ABI50_0_0RCTViewManager)
ABI50_0_0RCT_EXPORT_VIEW_PROPERTY(onFormComplete, ABI50_0_0RCTDirectEventBlock)
ABI50_0_0RCT_EXPORT_VIEW_PROPERTY(dangerouslyGetFullCardDetails, BOOL)
ABI50_0_0RCT_EXPORT_VIEW_PROPERTY(autofocus, BOOL)
ABI50_0_0RCT_EXPORT_VIEW_PROPERTY(cardStyle, NSDictionary)
ABI50_0_0RCT_EXPORT_VIEW_PROPERTY(disabled, BOOL)
ABI50_0_0RCT_EXTERN_METHOD(focus:(nonnull NSNumber*) ABI50_0_0ReactTag)
ABI50_0_0RCT_EXTERN_METHOD(blur:(nonnull NSNumber*) ABI50_0_0ReactTag)
@end

File: ios/vendored/sdk50/@stripe/stripe-react-native/ios/CardFormManager.swift
import Foundation

@objc(ABI50_0_0CardFormManager)
class CardFormManager: ABI50_0_0RCTViewManager {
    override func view() -> UIView! {
        let cardForm = CardFormView()
        let stripeSdk = bridge.module(forName: "StripeSdk") as? StripeSdk
        stripeSdk?.cardFormView = cardForm;
        return cardForm
    }
    
    override class func requiresMainQueueSetup() -> Bool {
        return false
    }
    
    @objc func focus(_ abi50_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI50_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFormView = (viewRegistry![abi50_0_0ReactTag] as? CardFormView)!
            view.focus()
        }
    }
    
    @objc func blur(_ abi50_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI50_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFormView = (viewRegistry![abi50_0_0ReactTag] as? CardFormView)!
            view.blur()
        }
    }
}
